vowpalwabbit/core/src/reductions/cb/details/large_action_space.h

```diff
@@ -102,5 +176,11 @@ void triplet_construction(TripletType& tc, float feature_value, uint64_t feature
   tc.set(feature_value, feature_index);
 }
 
+std::pair<float, uint64_t> find_max_volume(Eigen::MatrixXf& U, uint64_t x_row, Eigen::MatrixXf& X);
```

Comment:
why are these not member functions?
for example, `find_max_volume` could be a private of `cb_explore_adf_large_action_space` or even spanner state.

Reply:
I went with not member function to limit the scope of what info they have access to. I prefer less access and more because it interacted with U matrix.

Reply:
Sounds good. I think `find_max_volume` can still be a member of `spanner_state` though, since it is only used in `compute_spanner`.

Reply:
done